infra/modules/grafana-agent/main.tf

```diff
@@ -1,128 +1,16 @@
-                  basic_auth:
-                    username: ${var.grafana_tempo_username}
-                    password: $${GRAFANA_PASSWORD}
-              receivers:
-                otlp:
-                  protocols:
-                    http:
-                      cors:
-                        allowed_origins:
-                          - "http://*"
-                          - "https://*"
-      extraArgs:
-        - -config.expand-env
-      envFrom:
-        - secretRef:
-            name: grafana-password
-      extraPorts:
-        - name: otelhttp
-          port: 4318
-          targetPort: 4318
-          protocol: "TCP"
-  VALUES
-  ]
-}
-
-resource "helm_release" "grafana-k8s-monitoring" {
-  name       = "grafana-k8s-monitoring"
-  repository = "https://grafana.github.io/helm-charts"
-  chart      = "k8s-monitoring"
-  version    = "0.1.4"
-  namespace  = var.namespace
+  source  = "aws-ia/eks-blueprints-addon/aws"
+  version = "1.1.1"
 
-  values = [<<-VALUES
-    cluster:
-      name: ${var.cluster_name}
-    externalServices:
-      prometheus:
-        host: https://prometheus-us-central1.grafana.net
-        basicAuth:
-          username: ${var.grafana_prom_username}
-          password: ${data.kubernetes_secret.api_key_secret.data["GRAFANA_PASSWORD"]}
-      loki:
-        host: https://logs-prod-017.grafana.net
-        basicAuth:
-          username: ${var.grafana_loki_username}
-          password: ${data.kubernetes_secret.api_key_secret.data["GRAFANA_PASSWORD"]}
-    metrics:
-      cost:
-        enabled: false
-      podMonitors:
-        namespaces:
-          - "marlowe-staging"
-          - "marlowe-production"
-          - "dapps-certification-staging"
-          - "dapps-certification"
-      probes:
-        namespaces:
-          - "marlowe-staging"
-          - "marlowe-production"
-          - "dapps-certification-staging"
-          - "dapps-certification"
-      serviceMonitors:
-        namespaces:
-          - "marlowe-staging"
-          - "marlowe-production"
-          - "dapps-certification-staging"
-          - "dapps-certification"
-    opencost:
-      enabled: false 
-    crds:
-      create: false
-    logs:
-      pod_logs:
-        namespaces:
-          - "marlowe-staging"
-          - "marlowe-production"
-          - "dapps-certification-staging"
-          - "dapps-certification"
-      cluster_events:
-        namespaces:
-          - "marlowe-staging"
-          - "marlowe-production"
-          - "dapps-certification-staging"
-          - "dapps-certification"
-  VALUES
-  ]
-}
+  chart            = local.chart
+  chart_version    = local.chart_version
+  repository       = local.repository
+  description      = local.description
+  namespace        = local.namespace
+  create_namespace = local.create_namespace
+  set              = local.grafana_agent.set
+  values           = try(var.grafana_agent.values, local.grafana_agent.values)
```

Comment:
I think you could avoid this `try` by setting a default value for `var.grafana_agent.values` and then getting rid of the local.

Reply:
Yeah, I did it this way to have this module "agnostic" (without any default values related to our current work) so if someone (from other tribes) wants to use this module, it's just passing the needed inputs.